### PR TITLE
build: update libcurl to curl-8_10_1-241-g461ce6c61

### PR DESCRIPTION
**Pull request by Totktonada**
The reason is that the previous libcurl submodule update in commit 0919f390802f146852b462215327ef03e2730cfc ("third_party: update libcurl from 8.8.0 to 8.10.1") reveals the following regression:

```c
$ tarantool -e "require('http.client').new():get('https://google.com')"
tarantool: ./third_party/curl/lib/multi.c:3691: curl_multi_assign: Assertion `!(multi)' failed.
Aborted (core dumped)
```

The stacktrace is the following:

```c
<...>
#4  __assert_fail
#5  curl_multi_assign            // <- called by us
#6  curl_multi_sock_cb           // <- this is our callback
#7  Curl_multi_pollset_ev
#8  cpool_update_shutdown_ev
#9  cpool_discard_conn
#10 cpool_close_and_destroy_all
#11 Curl_cpool_destroy
#12 curl_multi_cleanup
#13 curl_env_finish              // <- destroy the multi handle
#14 httpc_env_finish
#15 luaT_httpc_cleanup
#16 lj_BC_FUNCC
#17 gc_call_finalizer
#18 gc_finalize
#19 gc_onestep
#20 lj_gc_fullgc
#21 lua_gc
#22 lj_cf_collectgarbage
#23 lj_BC_FUNCC
#24 lua_pcall
#25 luaT_call
#26 lua_main
#27 run_script_f
#28 fiber_cxx_invoke
#29 fiber_loop
#30 coro_init
```

The multi handle is during the destroy, but our `CURLMOPT_SOCKETFUNCTION` callback is invoked and the `curl_multi_assign()` call (invoked to associate a libev watcher to the given file descriptor) fails on the assertion. Everything is as described in https://github.com/curl/curl/issues/15201.

The first bad libcurl's commit is [curl-8_10_0-4-g48f61e781][1], but later it was fixed in [curl-8_10_1-241-g461ce6c61][2]. This commit updates libcurl to this revision to fix the regression.

Adjusted build options in our build script:

* Added `CURL_DISABLE_IPFS=ON`: [curl-8_10_1-57-gce7d0d413][3]
* Added `CURL_TEST_BUNDLES=OFF`: [curl-8_10_1-67-g71cf0d1fc][4]
* Changed `ENABLE_WEBSOCKETS=OFF` to `CURL_DISABLE_WEBSOCKETS=ON`: [curl-8_10_1-130-gd78e129d5][5]

[1]: https://github.com/curl/curl/commit/48f61e781a01e6a8dbc4a347e280644b1c68ab6a
[2]: https://github.com/curl/curl/commit/461ce6c6160b86439ddd74c59541231ec9e8558e
[3]: https://github.com/curl/curl/commit/ce7d0d41378007eda676c83ad6b86c59870cc9f1
[4]: https://github.com/curl/curl/commit/71cf0d1fca9e1f53524e1545ef0c08d174458d80
[5]: https://github.com/curl/curl/commit/d78e129d50b2d190f1c1bde2ad1f62f02f152db0

### CODE REVIEW

**Comment by Totktonada** [comment]
The CI job that verifies integration with `etcd-client` fails due to a reason irrelevant to this patch, see https://github.com/tarantool/etcd-client/issues/85.

File: changelogs/unreleased/bump-libcurl-to-8.10.1.md
Comment (p7nov) [inline changelogs/unreleased/bump-libcurl-to-8.10.1.md:3]: `+patches` hardly adds meaning.
If this change of build options affects users somehow, we better write the full sentence:
`Updated libcurl and patched its build options`
If it's just internal kitchen, then we better leave the previous text to avoid questions and confusion.

**Review by Col-Waltz — APPROVED** [review]
As I see by my local test this curl update fixes etcd tests. BTW how do we deal with failed tests for this commit if they are blocking merge in our etcd-client commit and here?

File: changelogs/unreleased/bump-libcurl-to-8.10.1.md
Comment (ligurio) [reply changelogs/unreleased/bump-libcurl-to-8.10.1.md:3]: Probably it's better to specify an exact version (`curl-8_10_1-241-g461ce6c61`) to avoid confusion with `+ patches` in changelog entry.

Comment (Totktonada) [reply changelogs/unreleased/bump-libcurl-to-8.10.1.md]: @p7nov Fair point!

@Col-Waltz, @ligurio Thanks for the suggestions! I think that an explicit `git describe` version is the best option. Updated this way.

File: cmake/BuildLibCURL.cmake
Comment (ligurio) [inline cmake/BuildLibCURL.cmake:214]: These cmake options are not related to a fix in libcurl. Could you move these changes to a separate commit?

**Comment by Totktonada** [comment]
> BTW how do we deal with failed tests for this commit if they are blocking merge in our etcd-client commit and here?

@Col-Waltz We have to ignore the known failure either here or in the etcd-client. I think that it depends on what will be ready to push first.

File: cmake/BuildLibCURL.cmake
Comment (Totktonada) [reply cmake/BuildLibCURL.cmake:214]: These options are added or changed within the given range of curl's commits.

`cmake -LAH` difference:

```diff
--- curl-a.txt	2024-10-14 14:35:07.607066031 +0300
+++ curl-b.txt	2024-10-14 14:35:15.239066096 +0300
@@ -9,7 +9,7 @@
 BUILD_CURL_EXE:BOOL=ON
 
 // Build libcurl examples
-BUILD_EXAMPLES:BOOL=OFF
+BUILD_EXAMPLES:BOOL=ON
 
 // Build libcurl man pages
 BUILD_LIBCURL_DOCS:BOOL=ON
@@ -239,7 +239,7 @@
 // Path to the CA bundle has been set
 CURL_CA_BUNDLE_SET:BOOL=TRUE
 
-// Path to the CA bundle to embed into the curl tool.
+// Path to the CA bundle to embed in the curl tool.
 CURL_CA_EMBED:STRING=
 
 // Set ON to use built-in CA store of TLS backend. Defaults to OFF
@@ -311,6 +311,9 @@
 // Disable installation targets
 CURL_DISABLE_INSTALL:BOOL=OFF
 
+// Disable IPFS
+CURL_DISABLE_IPFS:BOOL=OFF
+
 // Disable Kerberos authentication
 CURL_DISABLE_KERBEROS_AUTH:BOOL=OFF
 
@@ -383,6 +386,9 @@
 // Disable verbose strings
 CURL_DISABLE_VERBOSE_STRINGS:BOOL=OFF
 
+// Disable WebSockets
+CURL_DISABLE_WEBSOCKETS:BOOL=OFF
+
 // Enable CMake export target
 CURL_ENABLE_EXPORT_TARGET:BOOL=ON
 
@@ -401,6 +407,9 @@
 // Enable compiler Link Time Optimizations
 CURL_LTO:BOOL=OFF
 
+// Bundle libtests/unittests into single binaries
+CURL_TEST_BUNDLES:BOOL=OFF
+
 // Enable BearSSL for SSL/TLS
 CURL_USE_BEARSSL:BOOL=OFF
 
@@ -473,9 +482,6 @@
 // Enable Unix domain sockets support
 ENABLE_UNIX_SOCKETS:BOOL=ON
 
-// Enable WebSockets (experimental)
-ENABLE_WEBSOCKETS:BOOL=OFF
-
 // Path to a program.
 HTTPD:FILEPATH=HTTPD-NOTFOUND
 
@@ -545,7 +551,7 @@
 // Enable librtmp from rtmpdump
 USE_LIBRTMP:BOOL=OFF
 
-// Use msquic library for HTTP/3 support
+// Use msh3/msquic library for HTTP/3 support
 USE_MSH3:BOOL=OFF
 
 // Use nghttp2 library
```

Comment (ligurio) [reply cmake/BuildLibCURL.cmake:214]: > These options are added or changed within the given range of curl's commits.

Got it, thanks!

File: changelogs/unreleased/bump-libcurl-to-8.10.1.md
Comment (ligurio) [reply changelogs/unreleased/bump-libcurl-to-8.10.1.md]: Thanks!

**Review by ligurio — APPROVED** [review]
Thanks for the fix! LGTM after passing etcd-client tests.

**Comment by Totktonada** [comment]
Backported to `release/3.2` in PR #10685.

Backported to `release/2.11` in PR #10686.

**Pull request merged**
